packages/react-instantsearch/src/core/indexUtils.js

```diff
@@ -125,13 +133,25 @@ export function getCurrentRefinementValue(
   refinementsCallback
 ) {
   const index = getIndex(context);
+  const namespace = getNamespaceAndId(id);
   const refinements =
     (hasMultipleIndex(context) && has(searchState, `indices.${index}.${id}`)) ||
+    (!hasMultipleIndex(context) &&
+      namespace.namespace &&
+      namespace.id &&
+      has(searchState[namespace.namespace], namespace.id)) ||
     (!hasMultipleIndex(context) && has(searchState, id));
   if (refinements) {
-    const currentRefinement = hasMultipleIndex(context)
-      ? get(searchState.indices[index], id)
-      : get(searchState, id);
+    let currentRefinement;
+
+    if (hasMultipleIndex(context)) {
+      currentRefinement = get(searchState.indices[index], id);
```

Comment:
This will still not work for multi indices application. We need to replace by something like: has(
        searchState.indices[`${index}`][namespace.namespace],
        `${namespace.id}`
      ))